velox/functions/prestosql/tests/DecimalArithmeticTest.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <optional>
+#include "velox/functions/prestosql/tests/FunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::functions::test;
+
+namespace {
+
+class DecimalArithmeticTest : public FunctionBaseTest {
+ protected:
+  template <typename T>
+  void testDecimalExpr(
+      const VectorPtr& expected,
+      const std::string& expression,
+      const std::vector<VectorPtr>& input) {
+    VectorPtr result =
+        evaluate<SimpleVector<T>>(expression, makeRowVector(input));
+    assertEqualVectors(expected, result);
+    ASSERT_EQ(result->type()->toString(), expected->type()->toString());
+  }
+
+  void testDecimalAdd() {
+    auto resultLongFlat = makeDecimalFlatVector<LongDecimal>(
+        {LongDecimal(2000), LongDecimal(4000)}, 19, 3);
+    auto shortFlat = makeDecimalFlatVector<ShortDecimal>(
+        {ShortDecimal(1000), ShortDecimal(2000)}, 18, 3);
+    // Add short and short, returning long.
+    testDecimalExpr<LongDecimal>(
+        resultLongFlat, "decimal_add(c0, c1)", {shortFlat, shortFlat});
+    // Add short and long, returning long.
+    auto longFlat = makeDecimalFlatVector<LongDecimal>(
+        {LongDecimal(1000), LongDecimal(2000)}, 19, 3);
+    resultLongFlat = makeDecimalFlatVector<LongDecimal>(
+        {LongDecimal(2000), LongDecimal(4000)}, 20, 3);
+    testDecimalExpr<LongDecimal>(
+        resultLongFlat, "decimal_add(c0, c1)", {shortFlat, longFlat});
+    // Add short and long, returning long.
+    testDecimalExpr<LongDecimal>(
+        resultLongFlat, "decimal_add(c0, c1)", {longFlat, shortFlat});
+
+    // Add long and long, returning long.
+    testDecimalExpr<LongDecimal>(
+        resultLongFlat, "decimal_add(c0, c1)", {longFlat, longFlat});
```

Comment:
These tests seem to cover only flat-flat branch. Let's add tests for all other code paths.